### DISCUSSION TITLE
竖着的合并单元格会识别错误。

### DISCUSSION THREAD

**Discussion by 102757017** [RapidAI/TableStructureRec#71] [General]
竖着的合并单元格会被识别成了几个单独的格子，里面的文本也被拆分到不同格子了，如下图所示。
这种情况除了对模型迁移训练，还有其它调整方法吗
![image](https://github.com/user-attachments/assets/7dbd4c84-a54c-42a3-b5b9-f86e1c5bd3f5)

**Comment by Joker1212** [comment]
> 竖着的合并单元格会被识别成了几个单独的格子，里面的文本也被拆分到不同格子了，如下图所示。 这种情况除了对模型迁移训练，还有其它调整方法吗 ![image](https://private-user-images.githubusercontent.com/30728579/385100711-7dbd4c84-a54c-42a3-b5b9-f86e1c5bd3f5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzEzNzU1NzAsIm5iZiI6MTczMTM3NTI3MCwicGF0aCI6Ii8zMDcyODU3OS8zODUxMDA3MTEtN2RiZDRjODQtYTU0Yy00MmEzLWI1YjktZjg2ZTFjNWJkM2Y1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMTIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTEyVDAxMzQzMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM2NmZlYjhlZDRlZWY1ZTZlYjBmYTYxMzU4NDRlMTRhZDJhZWY1NWI0Y2E4ZjkyY2U5NTk2ZmQ5NTA1YjRmNDMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.KthuGFvxRZmV7eeozTgtTgp478nA_puTKLN1O67tjp4)


能麻烦给一下这部分的原图吗？内部有一些参数，但是未对外提供参数，可能需要依靠源码调整，我可以先试试是否可行

**Reply by zaoa0** [reply]
同样问题
![屏幕截图 2024-11-11 154331](https://github.com/user-attachments/assets/99a1ffc7-4c36-435e-8f10-a984f05038ff)
![image](https://github.com/user-attachments/assets/6cb153ca-e99a-4494-a5ed-69beb49213fc)
![屏幕截图 2024-11-11 160427](https://github.com/user-attachments/assets/74efaed0-f664-4c0c-bc5c-8245928a598e)
![image](https://github.com/user-attachments/assets/b36609d0-ab12-4ce2-a855-9142e70877b9)

**Comment by Joker1212** [comment]
emmm,大概知道为什么了，是代码执行时为了找到更多的横竖线和识别框，进行了闭合运算还有线延长，我可以加参数让大家自己决定

**Reply by 102757017** [reply]
看了下源码，是不是要修改utils_table_line_rec.py文件中下面这个函数
def adjust_lines(lines, alph=50, angle=50):
这里的alph是两个线段的端点之间的距离小于 alph，就将它们连接起来吗，alph有没有区分水平方向与垂直方向？

**Reply by Joker1212** [reply]
嗯，还有前面的地方，我已经把参数弄出来了，下一版发了，直接在调用时关闭就行了，可以参考 add_param_for_ocr 分支，如果紧急的话

**Reply by Joker1212** [reply]
已经在该分支readme说明了，使用方式

**Reply by Joker1212** [reply]
<img width="633" alt="image" src="https://github.com/user-attachments/assets/403ee83d-db28-425f-bda9-a48a3954731e">

**Comment by Joker1212** [comment]
已经合并代码了哈，版本也更新了，可能清华的源同步需要等等，可以参考参数进行设置了，本场景，建议把增强线框搜索的参数都设置为Fasle
